assets/src/blocks/Media/MediaBlock.js

```diff
@@ -17,6 +17,16 @@ const attributes = {
   },
 };
 
+// Old content didn't store the poster url and embed html in the props. For the frontend this is caught in the block's
+// backend render method, where we setup a div to be frontend rendered by the same MediaFrontend component as is used for
+// save.
+export const lacksAttributes = attributes => {
```

Comment:
Isn't this (and `patchLegacyAttributes` below) introducing more backwards compatibilty code than the current version? It looks pretty version-specific too, I think it's even harder to adapt or remove.

Reply:
It may be more lines, but they're much easier to remove than the effect:
https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/compare/media-change-listener...media-change-listener-remove-bc

Reply:
Also with this PR the BC code only affects content that actually needs it, while keeping our render code clean and not performing updates of the block's attributes, which is not a render concern.

Reply:
> It looks pretty version-specific too

Isn't BC code by nature version specific? :smile:

Reply:
Ha, I guess, but some can be more specific than others. So what's the next step here? Remove the WIP label?

Reply:
Sure, I'll just rebase it :+1: